li_chao_tree_sparse.cpp

```diff
@@ -0,0 +1,41 @@
+template <typename T> struct Line {
+    T m, b;
+    Line(T m0, T b0) : m(m0), b(b0) {}
+    T eval(T x) { return x * m + b; }
+};
+using line = Line<ll>; // Define!!!
+const ll MINX = -1e6, MAXX = 1e6; // Define!!!
+struct LiChao { // Just *-1 if you want to do max lol
+    struct Node {
+        Node *l = nullptr, *r = nullptr;
+        line val{0, LLINF};
```

Comment:
`LLINF` is not currently defined anywhere. Maybe it should be a constructor argument for `LiChao`?